sql/sql_connect.cc

```diff
@@ -255,6 +261,125 @@ void time_out_user_resource_limits(THD *thd, USER_CONN *uc)
   DBUG_VOID_RETURN;
 }
 
+static uint64 get_connection_delay_time(uint32 count)
+{
+  /*
+   * Based on that invalid password errors will cause user can't connect server
+  ever and server close connection if it got message from client after
+  wait_timeout
+   * so ,
+   step = timeout/(max_password_errors - failed_attempts_before_delay)
+  delay = step *(failed_count  -failed_attempts_before_delay)
+*/
+  struct system_variables *p= &global_system_variables;
+  /* Time unit for delay step is seconds , convert to milliseconds*/
+  uint64 timeout= (p->net_wait_timeout < p->net_interactive_timeout
+                       ? p->net_wait_timeout
+                       : p->net_interactive_timeout) *
+                  1000;
+  uint64 step= timeout / (max_password_errors - p->failed_attempts_before_delay);
+  DBUG_PRINT("info",("The step for connection delay is %d milliseconds",step));
+  /* min  delay time is 1 seconds and max delay time is timeout */
+  uint64 delay=
+      (step > 1000 ? step : 1000) * (count - p->failed_attempts_before_delay);
```

Comment:
why do you cap the value with 1 second?

Reply:
I have calculated this case :  if max_password_errors is UNIT_MAX  and timeout is 24h , delay time is nearly ```=24*3600*1000/40_0000_0000=24*36/40000=216/10000=0.0216``` ms . The result is too small , which influence nothing.

Reply:
Okay. I personally wouldn't do it, but up to you. Just, please, explain this logic in a comment.